### PR TITLE
chore: add `organizer` data structure for docker folders

### PR DESCRIPTION
**Pull request by pujitm**
## Summary by CodeRabbit

* **New Features**
  * Introduced a comprehensive validation system for organizer data, including structural and referential integrity checks for views and organizers.
  * Added new data models for resources, folders, references, and views, with strong typing and validation.
  * Implemented a sequential validation processor with configurable fail-fast behavior and detailed error reporting.
  * Added a dedicated service for managing and validating Docker organizer configuration files.

* **Bug Fixes**
  * Corrected spelling of error-related properties from "errorOccured" to "errorOccurred" in multiple services to ensure consistent error handling.

* **Tests**
  * Added extensive unit tests for organizer validation logic, view structure validation, and the validation processor to ensure correctness across various edge cases and scenarios.
  * Added comprehensive tests verifying validation processor behavior under diverse conditions.

### CODE REVIEW

File: api/src/unraid-api/organizer/organizer.validation.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * Checks if all resources referenced by the view are present in the resources set.
+ *
+ * @param view - The view to check.
+ * @param resources - The set of resources to check against.
+ * @returns True if all resources referenced by the view are present in the resources set, false otherwise.
+ */
+export function validateViewResourceRefs(view: OrganizerView, resources: Set<string>): boolean {
+    const refs = getRefsFromViewEntries(view.entries);
+    return resources.isSupersetOf(refs);
+}
```
[the remark this answers is not in the document]
Comment (pujitm) [reply api/src/unraid-api/organizer/organizer.validation.ts:53]: the api is built with vite, which provides this i think

[the remark this answers is not in the document]
Comment (pujitm) [reply api/src/unraid-api/organizer/organizer.validation.ts:53]: we're running node v22.17.0 in production, which supports it:
```bash
root@zima:~# node
Welcome to Node.js v22.17.0.
Type ".help" for more information.
> let a = new Set()
undefined
> a.isSupersetOf
[Function: isSupersetOf]
> .exit
```

we'll never be targeting an older runtime, so this seems unnecessary

File: api/src/core/utils/validation/validation-processor.test.ts [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+            type ComplexInput = {
+                email: string;
+                age: number;
+                tags: string[];
+            };
+
+            const steps = [
+                {
+                    name: 'emailFormat',
+                    validator: (input: ComplexInput) =>
```
Comment (elibosley) [inline api/src/core/utils/validation/validation-processor.test.ts:246]: Is this kind of thing something we should be using class-validator for? Or is this more specialized?

File: api/src/core/utils/validation/validation-processor.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
+/**
+ * @fileoverview Type-safe sequential validation processor
```
Comment (elibosley) [inline api/src/core/utils/validation/validation-processor.ts:2]: Yeah, my only question here is should we have utilized a library to provide the type-safe validations, and then add the chaining logic ourselves? Seems like it could be a case of not-built-here but it may be 100% necessary

Comment (elibosley) [reply api/src/core/utils/validation/validation-processor.ts:2]: Here's what AI said: 🛠️ TL;DR Recommendation
Use Zod or Valibot if you want standard schema-based validation.

Stick with your custom createValidationProcessor if:

You want step-named error reporting

You need typed per-step errors

You want fail-fast customization

You want full flexibility over validator logic

File: api/src/unraid-api/auth/auth.service.ts [hunk 4]
```diff
@@ -118,7 +118,7 @@ export class AuthService {
                 }))
             );
 
-            const { errors, errorOccured } = await batchProcess(
+            const { errors, errorOccurred: errorOccured } = await batchProcess(
```
Comment (elibosley) [inline api/src/unraid-api/auth/auth.service.ts:121]: Suggestion for later - should we move to https://www.npmjs.com/package/p-map instead of the custom batchProcess - just thinking about developer maintainability and documentation as that library is essentially the same functionality but with a bit more control and options.

File: api/src/unraid-api/organizer/organizer-view.validation.test.ts [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+            root1: createFolder('root1', 'Root', []),
+        });
+
+        const result = findDuplicateChildren(view);
+        expect(result).toEqual([]);
+    });
+});
+
+describe('findCycleInView', () => {
+    it('should return null when no cycle exists', () => {
```
Comment (elibosley) [inline api/src/unraid-api/organizer/organizer-view.validation.test.ts:191]: This is really cool.

File: api/src/unraid-api/organizer/organizer-view.validation.test.ts [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+    it('should detect direct self-reference cycle', () => {
+        const view = createView('view1', 'Test View', 'root1', {
+            root1: createFolder('root1', 'Root', ['root1']),
+        });
+
+        const result = findCycleInView(view);
+        expect(result).toEqual(['root1', 'root1']);
+    });
+
+    it('should detect simple two-node cycle', () => {
```
Comment (elibosley) [inline api/src/unraid-api/organizer/organizer-view.validation.test.ts:212]: This too, you've literally built a file system you weren't kidding!

File: api/src/unraid-api/organizer/organizer-view.validation.ts [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        }
+    });
+
+    return missing;
+}
+
+/**
+ * Finds duplicate children in folder entries.
+ */
+export function findDuplicateChildren(
```
Comment (elibosley) [inline api/src/unraid-api/organizer/organizer-view.validation.ts:48]: Would we be better suited with a drop-in lib like: https://github.com/dagrejs/graphlib/wiki instead of building these data structures by hand? 

Alternatively, I see a world where we publish this as a separate library on our org, this is very cool code.

File: api/src/unraid-api/organizer/organizer.dto.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    IsDefined,
+    IsIn,
+    IsInt,
+    IsNotEmpty,
+    IsNumber,
+    IsObject,
+    IsOptional,
+    IsString,
+    ValidateNested,
+} from 'class-validator';
```
Comment (elibosley) [inline api/src/unraid-api/organizer/organizer.dto.ts:16]: Oh ! You are using class validator for this...

**Review by elibosley — APPROVED** [review]
This is really awesome, just had some thoughts about possibly leveraging some OOTB solutions for common patterns.

File: api/src/core/utils/validation/validation-processor.ts (hunk 3 above)
Comment (pujitm) [reply api/src/core/utils/validation/validation-processor.ts:2]: ik it's kinda crazy that the zod's out there aren't great for validating trees/graphs. the more convincing benefit is the fact that we can unit test validation steps + pipelines individually as opposed to the zod `refine` style.

File: api/src/unraid-api/auth/auth.service.ts (hunk 4 above)
Comment (pujitm) [reply api/src/unraid-api/auth/auth.service.ts:121]: 💯 that looks perfect
